lib/elixir/lib/map.ex

```diff
@@ -77,7 +77,9 @@ defmodule Map do
   """
   @spec new(Enum.t) :: map
   def new(enumerable) do
-    Enum.reduce(enumerable, %{}, fn {k, v}, acc -> put(acc, k, v) end)
+    enumerable
+    |> Enum.to_list
```

Comment:
Honestly I sort of wish that Map.new and MapSet.new didn't have to handle enumerables, and instead we just used Enum.into for those cases. However at this point it isn't worth the breaking change.